src/main/java/jenkins/branch/PrimaryBranchHealthMetric.java

```diff
@@ -111,7 +116,23 @@ public void observe(Item item) {
                     && ((Actionable) item).getAction(PrimaryInstanceMetadataAction.class) != null) {
                 if (reports == null) {
                     reports = new ArrayList<>();
-                    reports.add(getHealthReport(item));
+                    reports.addAll(Util.fixNull(getHealthReports(item)));
+                }
+            }
+        }
+
+        public static List<HealthReport> getHealthReports(Item item) {
+            if (item instanceof Job) {
+                return ((Job<?,?>) item).getBuildHealthReports();
```

Comment:
Are there performance effects or changing this from querying a single item to querying a set of items?

Reply:
Should be faster as the single report needs top get the list first